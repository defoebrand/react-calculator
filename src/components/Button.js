import PropTypes from 'prop-types';

const clickMe = value => {
  alert(value);
};
const Button = ({ name }) => (
  <button type="button" onClick={() => { clickMe(name); }}>{name}</button>
);

Button.propTypes = {
  name: PropTypes.string,
};
Button.defaultProps = {
  name: 'x',
};

export default Button;
