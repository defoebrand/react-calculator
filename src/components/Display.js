import React from 'react';
import PropTypes from 'prop-types';

const style = {
  display: 'flex',
  justifyContent: 'flex-end',
  alignItems: 'center',
  backgroundColor: '#848592',
  height: '100px',
  color: 'white',
  fontWeight: 'bold',
  textAlign: 'right',
  paddingRight: '10px',
  fontSize: '45px',
};

const Display = ({ result }) => (
  <div className="displayContainer" style={style}>
    <p className="displayOutput">{result}</p>
  </div>
);

Display.propTypes = {
  result: PropTypes.string,
};
Display.defaultProps = {
  result: null,
};

export default Display;
