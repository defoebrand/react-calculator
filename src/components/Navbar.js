const Navbar = () => {
  const style = {
    width: '100vw',
    display: 'flex',
    justifyContent: 'space-around',
    alignItems: 'center',
    height: '50px',
    backgroundColor: '#fffddd',
  };
  const linkStyle = {
    textDecoration: 'none',
    fontSize: 'calc(15px + 1vw)',
    fontWeight: 'bold',
    cursor: 'pointer',
  };
  return (
    <div style={style}>
      <a style={linkStyle} href="/">Home </a>
      <a style={linkStyle} href="/App">Calculator</a>
      <a style={linkStyle} href="/Quote">Quote Of The Day</a>
    </div>
  );
};

export default Navbar;
