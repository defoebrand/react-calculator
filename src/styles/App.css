.displayContainer {
  width: 50vw;
  margin: 15px auto 0;
  text-align: right;
  border: 1px solid black;
  background-color: lightgray;
  padding: 0 15px;
}

.displayOutput {
  font-weight: 700;
  letter-spacing: 2px;
}

.buttonPanel {
  max-width: 50vw;
  margin: 0 auto;
  text-align: center;
  border: 1px solid black;
  background-color: gray;
  padding: 15px;
}

button {
  margin: 1px;
}