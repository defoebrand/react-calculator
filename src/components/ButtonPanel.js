import PropTypes from 'prop-types';
import Button from './Button';

const buttonList = {
  group1: ['AC', '+/-', '%', '/'],
  group2: ['7', '8', '9', 'x'],
  group3: ['4', '5', '6', '-'],
  group4: ['1', '2', '3', '+'],
  group5: ['0', '.', '='],
};

const Panel = ({ clickHandler }) => (

  <div className="buttonPanel">
    {
      Object.keys(buttonList).map(key => (
        buttonList[key].map(value => (
          <Button key={value} name={value} group={key} handleClick={clickHandler} />
        ))))
    }
  </div>
);

Panel.propTypes = {
  clickHandler: PropTypes.func,
};
Panel.defaultProps = {
  clickHandler: null,
};

export default Panel;
