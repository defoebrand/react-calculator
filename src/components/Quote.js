import Navbar from './Navbar';

const Quote = () => {
  const style = {
    display: 'flex',
    flexDirection: 'column',
    width: '90vw',
    margin: '25px auto 0',
  };
  return (
    <>
      <Navbar />
      <div style={style}>
        <h1>Math Magicians</h1>
        <h3>Welcome to our page!</h3>
        <p>
          Mathematics is not about numbers, equations, computations, or
          algorithms: it is about understanding. â€“William Paul Thurston
        </p>
      </div>
    </>
  );
};

export default Quote;
