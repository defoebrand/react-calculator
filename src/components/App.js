import React from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';

import calculate from '../logic/calculate';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
    this.handleClick.bind(this);
  }

    handleClick = buttonName => {
      const data = this.state;
      const result = calculate(data, buttonName);
      this.setState({
        total: result.total,
        next: result.next,
        operation: result.operation,
      });
    }

    render() {
      const { next, total, operation } = this.state;
      let display;
      if (next) {
        display = <Display result={`${next}`} />;
      } else {
        display = <Display result="Hello!" />;
      }
      if (operation) {
        display = <Display result={`${next} ${operation}`} />;
      }
      if (total) {
        display = <Display result={`${total} ${operation} ${next}`} />;
      }
      return (
        <>

          {display}
          <ButtonPanel clickHandler={e => this.handleClick(e.target.innerText)} />
        </>
      );
    }
}

export default App;
