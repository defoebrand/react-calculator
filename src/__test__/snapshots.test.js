import renderer from 'react-test-renderer';
import App from '../components/App';
import Button from '../components/Button';
import ButtonPanel from '../components/ButtonPanel';
import Display from '../components/Display';
import Home from '../components/Home';
import Navbar from '../components/Navbar';
import Quote from '../components/Quote';

it('renders App correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Button correctly', () => {
  const tree = renderer
    .create(<Button />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders ButtonPanel correctly', () => {
  const tree = renderer
    .create(<ButtonPanel />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Display correctly', () => {
  const tree = renderer
    .create(<Display />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it('renders Home correctly', () => {
  const tree = renderer
    .create(<Home />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Navbar correctly', () => {
  const tree = renderer
    .create(<Navbar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Quote correctly', () => {
  const tree = renderer
    .create(<Quote />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
