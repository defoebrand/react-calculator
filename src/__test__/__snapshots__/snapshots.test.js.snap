// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['renders App correctly 1'] = `
Array [
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fffddd",
        "display": "flex",
        "height": "50px",
        "justifyContent": "space-around",
        "width": "100vw",
      }
    }
  >
    <a
      href="/"
      style={
        Object {
          "cursor": "pointer",
          "fontSize": "calc(15px + 1vw)",
          "fontWeight": "bold",
          "textDecoration": "none",
        }
      }
    >
      Home 
    </a>
    <a
      href="/App"
      style={
        Object {
          "cursor": "pointer",
          "fontSize": "calc(15px + 1vw)",
          "fontWeight": "bold",
          "textDecoration": "none",
        }
      }
    >
      Calculator
    </a>
    <a
      href="/Quote"
      style={
        Object {
          "cursor": "pointer",
          "fontSize": "calc(15px + 1vw)",
          "fontWeight": "bold",
          "textDecoration": "none",
        }
      }
    >
      Quote Of The Day
    </a>
  </div>,
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "margin": "0 auto",
        "width": "80vw",
      }
    }
  >
    <p
      style={
        Object {
          "fontSize": "35px",
          "margin": "25px 25px 0 15px",
        }
      }
    >
      Let's do some math!
    </p>
    <div
      style={
        Object {
          "border": "1px solid black",
          "display": "flex",
          "flexDirection": "column",
          "margin": "25px auto 0",
          "minWidth": "700px",
        }
      }
    >
      <div
        className="buttonPanel"
        style={
          Object {
            "backgroundColor": "#d0d0d0",
            "display": "flex",
            "flexWrap": "wrap",
            "lineHeight": "100px",
          }
        }
      >
        <button
          className="button group1"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#dfdfdf",
              "width": "25%",
            }
          }
          type="button"
        >
          AC
        </button>
        <button
          className="button group1"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#dfdfdf",
              "width": "25%",
            }
          }
          type="button"
        >
          +/-
        </button>
        <button
          className="button group1"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#dfdfdf",
              "width": "25%",
            }
          }
          type="button"
        >
          %
        </button>
        <button
          className="button group1"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f49040",
              "width": "25%",
            }
          }
          type="button"
        >
          /
        </button>
        <button
          className="button group2"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#dfdfdf",
              "width": "25%",
            }
          }
          type="button"
        >
          7
        </button>
        <button
          className="button group2"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#dfdfdf",
              "width": "25%",
            }
          }
          type="button"
        >
          8
        </button>
        <button
          className="button group2"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#dfdfdf",
              "width": "25%",
            }
          }
          type="button"
        >
          9
        </button>
        <button
          className="button group2"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f49040",
              "width": "25%",
            }
          }
          type="button"
        >
          x
        </button>
        <button
          className="button group3"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#dfdfdf",
              "width": "25%",
            }
          }
          type="button"
        >
          4
        </button>
        <button
          className="button group3"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#dfdfdf",
              "width": "25%",
            }
          }
          type="button"
        >
          5
        </button>
        <button
          className="button group3"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#dfdfdf",
              "width": "25%",
            }
          }
          type="button"
        >
          6
        </button>
        <button
          className="button group3"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f49040",
              "width": "25%",
            }
          }
          type="button"
        >
          -
        </button>
        <button
          className="button group4"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#dfdfdf",
              "width": "25%",
            }
          }
          type="button"
        >
          1
        </button>
        <button
          className="button group4"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#dfdfdf",
              "width": "25%",
            }
          }
          type="button"
        >
          2
        </button>
        <button
          className="button group4"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#dfdfdf",
              "width": "25%",
            }
          }
          type="button"
        >
          3
        </button>
        <button
          className="button group4"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f49040",
              "width": "25%",
            }
          }
          type="button"
        >
          +
        </button>
        <button
          className="button group5"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#dfdfdf",
              "width": "50%",
            }
          }
          type="button"
        >
          0
        </button>
        <button
          className="button group5"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#dfdfdf",
              "width": "25%",
            }
          }
          type="button"
        >
          .
        </button>
        <button
          className="button group5"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f49040",
              "width": "25%",
            }
          }
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>,
]
`;

exports['renders Button correctly 1'] = `
<button
  className="button Empty"
  onClick={null}
  style={
    Object {
      "backgroundColor": "#f49040",
      "width": "25%",
    }
  }
  type="button"
>
  Empty
</button>
`;

exports['renders ButtonPanel correctly 1'] = `
<div
  className="buttonPanel"
  style={
    Object {
      "backgroundColor": "#d0d0d0",
      "display": "flex",
      "flexWrap": "wrap",
      "lineHeight": "100px",
    }
  }
>
  <button
    className="button group1"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "width": "25%",
      }
    }
    type="button"
  >
    AC
  </button>
  <button
    className="button group1"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "width": "25%",
      }
    }
    type="button"
  >
    +/-
  </button>
  <button
    className="button group1"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "width": "25%",
      }
    }
    type="button"
  >
    %
  </button>
  <button
    className="button group1"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#f49040",
        "width": "25%",
      }
    }
    type="button"
  >
    /
  </button>
  <button
    className="button group2"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "width": "25%",
      }
    }
    type="button"
  >
    7
  </button>
  <button
    className="button group2"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "width": "25%",
      }
    }
    type="button"
  >
    8
  </button>
  <button
    className="button group2"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "width": "25%",
      }
    }
    type="button"
  >
    9
  </button>
  <button
    className="button group2"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#f49040",
        "width": "25%",
      }
    }
    type="button"
  >
    x
  </button>
  <button
    className="button group3"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "width": "25%",
      }
    }
    type="button"
  >
    4
  </button>
  <button
    className="button group3"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "width": "25%",
      }
    }
    type="button"
  >
    5
  </button>
  <button
    className="button group3"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "width": "25%",
      }
    }
    type="button"
  >
    6
  </button>
  <button
    className="button group3"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#f49040",
        "width": "25%",
      }
    }
    type="button"
  >
    -
  </button>
  <button
    className="button group4"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "width": "25%",
      }
    }
    type="button"
  >
    1
  </button>
  <button
    className="button group4"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "width": "25%",
      }
    }
    type="button"
  >
    2
  </button>
  <button
    className="button group4"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "width": "25%",
      }
    }
    type="button"
  >
    3
  </button>
  <button
    className="button group4"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#f49040",
        "width": "25%",
      }
    }
    type="button"
  >
    +
  </button>
  <button
    className="button group5"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "width": "50%",
      }
    }
    type="button"
  >
    0
  </button>
  <button
    className="button group5"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#dfdfdf",
        "width": "25%",
      }
    }
    type="button"
  >
    .
  </button>
  <button
    className="button group5"
    onClick={null}
    style={
      Object {
        "backgroundColor": "#f49040",
        "width": "25%",
      }
    }
    type="button"
  >
    =
  </button>
</div>
`;

exports['renders Display correctly 1'] = `
<div
  className="displayContainer"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#848592",
      "color": "white",
      "display": "flex",
      "fontSize": "45px",
      "fontWeight": "bold",
      "height": "100px",
      "justifyContent": "flex-end",
      "paddingRight": "10px",
      "textAlign": "right",
    }
  }
>
  <p
    className="displayOutput"
  />
</div>
`;

exports['renders Home correctly 1'] = `
Array [
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fffddd",
        "display": "flex",
        "height": "50px",
        "justifyContent": "space-around",
        "width": "100vw",
      }
    }
  >
    <a
      href="/"
      style={
        Object {
          "cursor": "pointer",
          "fontSize": "calc(15px + 1vw)",
          "fontWeight": "bold",
          "textDecoration": "none",
        }
      }
    >
      Home 
    </a>
    <a
      href="/App"
      style={
        Object {
          "cursor": "pointer",
          "fontSize": "calc(15px + 1vw)",
          "fontWeight": "bold",
          "textDecoration": "none",
        }
      }
    >
      Calculator
    </a>
    <a
      href="/Quote"
      style={
        Object {
          "cursor": "pointer",
          "fontSize": "calc(15px + 1vw)",
          "fontWeight": "bold",
          "textDecoration": "none",
        }
      }
    >
      Quote Of The Day
    </a>
  </div>,
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "margin": "25px auto 0",
        "width": "90vw",
      }
    }
  >
    <h1>
      Math Magicians
    </h1>
    <h3>
      Welcome to our page!
    </h3>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
    </p>
  </div>,
]
`;

exports['renders Navbar correctly 1'] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fffddd",
      "display": "flex",
      "height": "50px",
      "justifyContent": "space-around",
      "width": "100vw",
    }
  }
>
  <a
    href="/"
    style={
      Object {
        "cursor": "pointer",
        "fontSize": "calc(15px + 1vw)",
        "fontWeight": "bold",
        "textDecoration": "none",
      }
    }
  >
    Home 
  </a>
  <a
    href="/App"
    style={
      Object {
        "cursor": "pointer",
        "fontSize": "calc(15px + 1vw)",
        "fontWeight": "bold",
        "textDecoration": "none",
      }
    }
  >
    Calculator
  </a>
  <a
    href="/Quote"
    style={
      Object {
        "cursor": "pointer",
        "fontSize": "calc(15px + 1vw)",
        "fontWeight": "bold",
        "textDecoration": "none",
      }
    }
  >
    Quote Of The Day
  </a>
</div>
`;

exports['renders Quote correctly 1'] = `
Array [
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fffddd",
        "display": "flex",
        "height": "50px",
        "justifyContent": "space-around",
        "width": "100vw",
      }
    }
  >
    <a
      href="/"
      style={
        Object {
          "cursor": "pointer",
          "fontSize": "calc(15px + 1vw)",
          "fontWeight": "bold",
          "textDecoration": "none",
        }
      }
    >
      Home 
    </a>
    <a
      href="/App"
      style={
        Object {
          "cursor": "pointer",
          "fontSize": "calc(15px + 1vw)",
          "fontWeight": "bold",
          "textDecoration": "none",
        }
      }
    >
      Calculator
    </a>
    <a
      href="/Quote"
      style={
        Object {
          "cursor": "pointer",
          "fontSize": "calc(15px + 1vw)",
          "fontWeight": "bold",
          "textDecoration": "none",
        }
      }
    >
      Quote Of The Day
    </a>
  </div>,
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "margin": "25px auto 0",
        "width": "90vw",
      }
    }
  >
    <h1>
      Math Magicians
    </h1>
    <h3>
      Welcome to our page!
    </h3>
    <p>
      Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. –William Paul Thurston
    </p>
  </div>,
]
`;
